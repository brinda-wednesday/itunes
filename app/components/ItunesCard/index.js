/**
 *
 * ItunesCard
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Card, Avatar } from 'antd';
import styled from 'styled-components';
import { primarygrey } from '@app/themes/colors';

const { Meta } = Card;

const CustomCard = styled(Card)`
  && {
    margin: 1.25em 1.25em;
    width: 25em;
    border-radius: 1.25em;
    background-color: ${primarygrey};
    position: relative;
  }
`;
const Audio = styled.audio`
  && {
    width: 18em;
    margin: 1.25em;
  }
`;
const Source = styled.source`
  && {
    background-color: ${primarygrey};
  }
`;
const CustomAvatar = styled(Avatar)`
  && {
    width: 5em;
    height: 5em;
  }
`;
export function ItunesCard({ songTitle, imgSrc, songArtist, audioSrc, trackId }) {
  return (
    <CustomCard data-testid="itunes-card" key={trackId}>
      <Meta
        data-testid="song-name"
        avatar={<CustomAvatar data-testid="artist-img" src={imgSrc} />}
        title={songTitle}
        description={songArtist}
      />

      <Audio controls>
        <Source data-testid="audio-src" src={audioSrc} type="video/mp4" />
      </Audio>
    </CustomCard>
  );
}

ItunesCard.propTypes = {
  songTitle: PropTypes.string.isRequired,
  songArtist: PropTypes.string.isRequired,
  imgSrc: PropTypes.string.isRequired,
  audioSrc: PropTypes.string.isRequired,

  trackId: PropTypes.number.isRequired
};

export default ItunesCard;
