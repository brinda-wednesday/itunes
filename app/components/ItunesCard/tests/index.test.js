/**
 *
 * Tests for ItunesCard
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderProvider, renderWithIntl } from '@utils/testUtils';
import ItunesCard from '../index';

describe('<ItunesCard />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<ItunesCard />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 ItunesCard component', () => {
    const { getAllByTestId } = renderWithIntl(<ItunesCard />);
    expect(getAllByTestId('itunes-card').length).toBe(1);
  });
  it('should render song title', () => {
    const { getAllByTestId } = renderProvider(<ItunesCard />);
    expect(getAllByTestId('song-name').length).toBe(1);
  });
  it('should render artist image', () => {
    const { getAllByTestId } = renderProvider(<ItunesCard />);
    expect(getAllByTestId('artist-img').length).toBe(1);
  });
  it('should render audio track', () => {
    const { getAllByTestId } = renderProvider(<ItunesCard />);
    expect(getAllByTestId('audio-src').length).toBe(1);
  });
  it('testing for correct track name prop values', () => {
    const { getByTestId } = renderProvider(<ItunesCard songTitle="Faded" />);
    expect(getByTestId('song-name').textContent).toBe('Faded');
  });
  it('testing for correct  audio src prop values', () => {
    const { getByTestId } = renderProvider(
      <ItunesCard audioSrc="https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview125/v4/36/16/36/36163607-50ce-5dec-846a-782467ea6b09/mzaf_529304218680119840.plus.aac.p.m4a" />
    );
    expect(getByTestId('audio-src').src).toBe(
      'https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview125/v4/36/16/36/36163607-50ce-5dec-846a-782467ea6b09/mzaf_529304218680119840.plus.aac.p.m4a'
    );
  });
});
