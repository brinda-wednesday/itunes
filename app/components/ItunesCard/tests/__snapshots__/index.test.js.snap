// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItunesCard /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ant-card ant-card-bordered ItunesCard__CustomCard-sijrgi-0 dbuYLV"
      data-testid="itunes-card"
    >
      <div
        class="ant-card-body"
      >
        <div
          class="ant-card-meta"
          data-testid="song-name"
        >
          <div
            class="ant-card-meta-avatar"
          >
            <span
              class="ant-avatar ant-avatar-circle ItunesCard__CustomAvatar-sijrgi-3 bdsWnH"
              data-testid="artist-img"
            >
              <span
                class="ant-avatar-string"
                style="opacity: 0;"
              />
            </span>
          </div>
        </div>
        <audio
          class="ItunesCard__Audio-sijrgi-1 eukMhx"
          controls=""
        >
          <source
            class="ItunesCard__Source-sijrgi-2 cRCZEq"
            data-testid="audio-src"
            type="video/mp4"
          />
        </audio>
      </div>
    </div>
  </div>
</body>
`;
